var/spack/repos/builtin/packages/tldd/package.py

```diff
@@ -0,0 +1,47 @@
+##############################################################################
+# Copyright (c) 2013-2018, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/spack/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+
+class Tldd(Package):
+    """A program similar to ldd(1) but showing the output as a tree."""
+
+    homepage = "https://gitlab.com/miscripts/tldd"
+    url      = "https://gitlab.com/miscripts/tldd"
+    git      = "https://gitlab.com/miscripts/tldd.git"
+
+    version('master', branch='master')
+
+    depends_on('pstreams')
+
+    def install(self, spec, prefix):
```

Comment:
this method shouldn't be necessary.

Reply:
It didn't work out of the box unless I specified the PREFIX variable. 
Or maybe are you suggesting something like what is done for [p7zip](https://github.com/spack/spack/blob/develop/var/spack/repos/builtin/packages/p7zip/package.py) for example? Make it a `MakefilePackage` at redefine the `install_targets()`?

Reply:
Even if I make it a `MakefilePackage` it still doesn't work out of the box, it tries to install to `/usr/local/bin`, which comes from the Makefile: `PREFIX := /usr/local`. 

From the Environment Variables section in the documentation for [MakeFilePackage](https://spack.readthedocs.io/en/latest/build_systems/makefilepackage.html?highlight=makefilepackage#environment-variables) I now understand better what the options are. Depending on how the Makefile is written, you can either set an environment variable; pass a parameter (the tldd case); or edit the Makefile.

As the `:=` assignment operator is used for PREFIX I need to pass a parameter to the `make install`.